.github/workflows/pymeasure_CI.yml

```diff
@@ -46,6 +46,47 @@ jobs:
         run: |
           echo "::add-matcher::.github/sphinx.json"
           make doctest SPHINXOPTS="-W --keep-going"
+
+  test_coverage:
+    name: Code Coverage on codecov
+    runs-on: "ubuntu-latest"
+    defaults:
+      run:
+        shell: bash -l {0}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Install pymeasure requirements
+        uses: mamba-org/setup-micromamba@v1
+        with:
+          environment-file: .github/pymeasure.yml
+          create-args: python=${{ matrix.python-version }}
+          cache-environment-key: py${{ matrix.python-version }}-${{ matrix.os }}-mamba-${{ env.CACHE_NUMBER }}-${{ hashFiles('environment.yml') }}
+          cache-downloads: false
+      - name: Python version
+        run: python --version
+      - name: Install Pymeasure
+        # If the pytest problem matcher stops working because of bad paths, do an editable install
+        run: pip install -e .[tests]  # editable for covtest
+      - name: Pymeasure version
+        run: python -c "import pymeasure;print(pymeasure.__version__)"
+      - name: Test for Coverage
+        # xvfb for graphical interface
+        run: |
+          echo "::add-matcher::.github/pytest.json"
+          xvfb-run -a pytest --cov=pymeasure --cov-report=xml
+      - name: Upload coverage to Codecov
```

Comment:
One thing I wonder: for the coverage badge, can codecov distinguish between a push to master and a PR, or does it show the latest uploaded coverage report (regardless of whether it is from a PR)?